### PR TITLE
xdsclient: NACK endpoint resources with zero weight

### PR DESCRIPTION
**Pull request by easwars**
This is the existing behavior in other languages already (and Envoy), and we are late to the party.

Fixes: https://github.com/grpc/grpc-go/issues/5456

Summary of changes:
- Change the unmarshaling routine in `xdsclient` to error out if it sees an endpoint resource with zero weight
- Change test routines to insert a weight of 1 in all endpoint resources that they use
- Fix existing tests to expect endpoints with weight 1
- Add a test for the zero endpoint case

RELEASE NOTES:
- xds: NACK endpoint resources with zero weight

### CODE REVIEW

File: xds/internal/xdsclient/xdsresource/unmarshal_eds.go [hunk 1]
```diff
@@ -134,9 +138,13 @@ func parseEDSRespProto(m *v3endpointpb.ClusterLoadAssignment, logger *grpclog.Pr
 			return EndpointsUpdate{}, fmt.Errorf("duplicate locality %s with the same priority %v", lidStr, priority)
 		}
 		localitiesWithPriority[lidStr] = true
+		endpoints, err := parseEndpoints(locality.GetLbEndpoints())
+		if err != nil {
+			return EndpointsUpdate{}, err
```
Comment (zasweq) [inline xds/internal/xdsclient/xdsresource/unmarshal_eds.go:143]: Please wrap this err with more information (see other errors in this function).

**Review by zasweq — APPROVED** [review]
LGTM outside of minor nit.

File: xds/internal/xdsclient/xdsresource/unmarshal_eds.go (hunk 1 above)
Comment (easwars) [reply xds/internal/xdsclient/xdsresource/unmarshal_eds.go:143]: `parseEndpoints` returns an error which has all the required information.
